wp-content/plugins/wporg-learn/inc/sensei.php

```diff
@@ -163,3 +163,60 @@ function wporg_fix_learning_mode_header_space() {
 	wp_add_inline_style( 'learning-mode-header-fix', $custom_styles );
 }
 add_action( 'sensei_course_learning_mode_load_theme', __NAMESPACE__ . '\wporg_fix_learning_mode_header_space' );
+
+/**
+ * Add script to count unique learners
+ */
+function wporg_learn_add_student_count_to_reports( $type ) {
+	if ( 'users' !== $type ) {
+		return; // Only show the count on the students report screen.
+	}
+
+	$from_date = \DateTime::createFromFormat( 'Y-m-d', $_GET['from_date'] ?? '', new \DateTimeZone( 'UTC' ) );
+	$to_date   = \DateTime::createFromFormat( 'Y-m-d', $_GET['to_date'] ?? '', new \DateTimeZone( 'UTC' ) );
+
+	global $wpdb;
+	$query = "SELECT COUNT(DISTINCT user_id)
+FROM $wpdb->comments
+WHERE comment_type = 'sensei_course_status'";
+
+	if ( $from_date ) {
+		$from_date->setTime( 0, 0, 0 );
+
+		$query .= " AND comment_date_gmt >= '" . $from_date->format( 'Y-m-d H:i:s' ) . "'";
+	}
+
+	if ( $to_date ) {
+		$to_date->setTime( 23, 59, 59 );
+
+		$query .= " AND comment_date_gmt <= '" . $to_date->format( 'Y-m-d H:i:s' ) . "'";
+	}
+
+	$student_count = $wpdb->get_var( $query ); //phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
```

Comment:
@pkevan Ben and I discussed this, and we feel ignoring the PHPCS rule here is ok, but we'd appreciate your opinion. I personally can't see how this could be attacked via SQL injection, but maybe I'm missing something.

Reply:
While it's true that the current code isn't vulnerable, that doesn't mean in the future it wouldn't be, so it's preferable to built in elements which would force it to not be.

For example, the `$from_date` variable assignment logic could be changed to grab from the `$_GET` variable directly or the appended `$query` concatenation logic changed in a way which did make it.

By adding some prepare statements, it would help avoid those scenarios, particularly when dealing with user input.

Reply:
@pkevan Thanks, that's a fair point, I will work on adding sanitization to the date variables, and change the SQL concatenation to use a prepare statement.